napari/utils/transforms/transform_utils.py

```diff
@@ -215,3 +230,66 @@ def shear_matrix_from_angle(angle, ndim=3, axes=(-1, 0)):
     matrix = np.eye(ndim)
     matrix[axes] = np.tan(np.deg2rad(90 - angle))
     return matrix
+
+
+def is_upper_triangular(matrix):
+    """Check if a matrix is upper triangular.
+
+    Parameters
+    ----------
+    matrix : np.ndarray
+        Matrix to be checked.
+
+    Returns
+    -------
+    bool
+        Whether matrix is upper triangular or not.
+    """
+    return np.allclose(matrix, np.triu(matrix))
+
+
+def is_lower_triangular(matrix):
+    """Check if a matrix is lower triangular.
+
+    Parameters
+    ----------
+    matrix : np.ndarray
+        Matrix to be checked.
+
+    Returns
+    -------
+    bool
+        Whether matrix is lower triangular or not.
+    """
+    return np.allclose(matrix, np.tril(matrix))
+
+
+def check_shear_triangular(matrix):
+    """Check if a matrix is triangular.
+
+    Raises an error if neither upper or lower triangular.
+    1-D arrays are taken to be upper triangular.
+
+    Parameters
+    ----------
+    matrix : np.ndarray
+        Matrix to be checked.
+
+    Returns
+    -------
+    bool
+        True if matrix upper triangular, False if not.
+    """
+    if np.array(matrix).ndim == 2:
+        if is_upper_triangular(matrix):
+            return True
+        elif is_lower_triangular(matrix):
+            return False
```

Comment:
I don't like this - given the docstring "Check if a matrix is triangular" I expect the function to return `True` for either an upper triangular or lower triangular matrix, and return `False` (or raise an error) if the matrix is not any kind of triangular.